pandera/api/pyspark/types.py

```diff
@@ -2,13 +2,27 @@
 
 from functools import lru_cache
 from typing import List, NamedTuple, Tuple, Type, Union
+from numpy import bool_ as np_bool
+from packaging import version
 
 import pyspark.sql.types as pst
 from pyspark.sql import DataFrame
 
+import pyspark
 from pandera.api.checks import Check
 from pandera.dtypes import DataType
 
+# pylint: disable=reimported
+# Handles optional Spark Connect imports for pyspark>=3.4 (if available)
+if version.parse(pyspark.__version__) >= version.parse("3.4"):
+    from pyspark.sql.connect.dataframe import DataFrame as psc_DataFrame
+else:
+    from pyspark.sql import (
```

Comment:
in this case, wouldn't the Union type contain the same type? Since `psc_DataFrame` is imported from `pyspark.sql`

Reply:
Yes and this was the only way I figure out to be able to solve the typing annotation for a class that might or might not exists (since it'll be available only when using pyspark 3.4).
I'm relying exactly on this [behavior of `typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union):

> Redundant arguments are skipped, e.g.:
> ```py
> Union[int, str, int] == Union[int, str] == int | str
> ```

Reply:
cool, that sound fine